packages/notifications/push/amplify_push_notifications_pinpoint/lib/src/pinpoint_provider.dart

```diff
@@ -72,18 +80,21 @@ class PinpointProvider implements ServiceProviderClient {
   }) async {
     try {
       if (!_isInitialized) {
-        _logger.error(
-          'Pinpoint provider not configured, re-run Amplify.configure',
+        throw Exception(
+          'Pinpoint provider is not initialized. Re-run Amplify.configure.',
```

Comment:
This can't be generic `Exception` IMO
We need proper error message, recovery suggestion and underlying exception (if any).

In addition "re-run" is not accurate, Amplify doesn't support re-configure, so when the exception happens, the only case is that configure has never been called.